### DISCUSSION TITLE
How to make guest time host time?

### DISCUSSION THREAD

**Discussion by moenodedev** [lima-vm/lima#2396] [Q&A] [answered]
When I leave a guest running overnight and ssh into the instance the next day the time is off. How do I make the guest use the host time without sync? Is there a file that can be mounted from macOS onto a linux guest?

**Comment by jandubois** [comment, marked as answer]
> Is there a file that can be mounted from macOS onto a linux guest?

No, there is not. Lima does check time against RTC and adjust the time inside the guest when it is more than 10s off.

This is off course not good enough for certain use-cases (e.g. Makefiles), and it may also not always work. We have several open issues about this.

The "proper" solution would be running an NTP server on the host and then syncing to that from inside the guest. Just needs somebody to spend the time to implement it.

**Reply by moenodedev** [reply]
Since I want to avoid internet requests I used this hack:

I have a shell function on my host to ssh into the container. Before ssh the host's date is written to a tmp file.
```bash
lima-instance() {
  date >/tmp/lima/host-date-output
  ssh "$instance"
}
```

Then when my shell config is processed on the guest the date will be set if that file exists.
```bash
newdate=$(cat "/tmp/lima/host-date-output" 2>/dev/null)

if [ -s "$newdate" ]; then
  sudo date -s "$newdate" &>/dev/null
fi
```

**Reply by afbjorklund** [reply]
Something simple like https://github.com/btfak/sntp could work, for a basic NTP service
